pkg/stores/sqlpartition/queryparser/selector.go

```diff
@@ -0,0 +1,965 @@
+/*
+Copyright 2014 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+/*
+This file is derived from
+https://github.com/kubernetes/apimachinery/blob/master/pkg/labels/selector.go
```

Comment:
Similar to the point above, I think it helps to point to the exact commit and be descriptive about changes.

Reply:
Sure